src/docdb/nodes.ts

```diff
@@ -152,4 +183,36 @@ export class DocDBDocumentNode implements INode {
 		arguments: [this],
 		title: ''
 	};
+}
+
+export class LoadMoreNode implements INode {
+	constructor(readonly iterator: QueryIterator<any>, readonly parentNode: DocDBCollectionNode) {
+	}
+
+	readonly id = `${this.parentNode.id}.LoadMore`;
+
+	readonly label = `Load More...`;
+
+	readonly contextValue = 'LoadMoreButton'
+
+	readonly command: Command = {
+		command: 'cosmosDB.loadMore',
+		arguments: [this],
+		title: ''
+	};
+
+	static async loadNextKElements(iterator: QueryIterator<any>, k: number): Promise<any> {
+		let elements = [], i: number = 0, hasMoreItems: boolean = false;
+		let current = await new Promise<any>((resolve, reject) => iterator.nextItem((err, result) => err ? reject(err) : resolve(result)));
+		while (current !== undefined && i < k) {
+			elements.push(current);
+			i++;
+			current = await new Promise<any>((resolve, reject) => iterator.nextItem((err, result) => err ? reject(err) : resolve(result)));
+		}
+		if (current !== undefined) {
+			hasMoreItems = true;
+			elements.push(iterator);
```

Comment:
That's weird to push 'iterator' onto elements (it's a different type, correct?). Hopefully when you move this to `DocDBCollectionNode` you won't have to do that (you could just store `iterator` as a private variable in the class)